riscv/csrs.cc

```diff
@@ -467,7 +467,10 @@ mstatus_csr_t::mstatus_csr_t(processor_t* const proc, const reg_t addr):
       | (proc->extension_enabled_const('S') && (proc->get_const_xlen() != 32) ? set_field((reg_t)0, MSTATUS_SXL, xlen_to_uxl(proc->get_const_xlen())) : 0)
 
 #ifdef RISCV_ENABLE_DUAL_ENDIAN
-      | (proc->get_mmu()->is_target_big_endian() ? MSTATUS_UBE | MSTATUS_SBE | MSTATUS_MBE : 0)
+      | (proc->get_mmu()->is_target_big_endian() ? (proc->extension_enabled_const('U') ? MSTATUS_UBE : 0)
+                                                 | (proc->extension_enabled_const('S') ? MSTATUS_SBE : 0)
+                                                 | MSTATUS_MBE
+                                                 : 0)
```

Comment:
The syntax is awkward here because of the ternary operator. What if you made a separate const `big_endian_bits` first, and then use that to make this ternary into a single line again?

Reply:
OK.